modules/core/src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -312,7 +312,7 @@ private[kafka] final class KafkaConsumerActor[F[_], K, V](
         val records = withRebalancing.records.keySetStrict
 
         val revokedFetches = revoked intersect fetches
-        val revokedNonFetches = revoked diff fetches
+        val revokedNonFetches = revoked diff revokedFetches
```

Comment:
I would appreciate a comment on this change in particular.

The change makes sense to me, logically: `revokedFetches` and `revokedNonFetches` add up to `revoked`. But I did not dig enough into the implications to be sure this is the intention. For instance: in which combination of pending records and pending fetches it will make a difference, and what is the consequence?

I thought I had left this out of the PR, but while it's here we might as well discuss it.